numpy/typing/_callable.py

```diff
@@ -312,6 +314,12 @@ def __call__(
     class _NumberOp(Protocol):
         def __call__(self, __other: _NumberLike) -> number: ...
 
+    class _ComparisonOp(Protocol[_T]):
+        @overload
+        def __call__(self, __other: _T) -> bool_: ...
+        @overload
+        def __call__(self, __other: ArrayLike) -> Union[ndarray, bool_]: ...
```

Comment:
The comparison ops of `generic` are one the few that accept actual array-like object (rather than just arrays and/or scalar).
Consequently, the signatures are still a bit too broad (something to revisit after https://github.com/numpy/numpy/pull/17719):

``` python
import numpy as np

U = np.str_()
i8 = np.int64()

i8 > U  # This is considered acceptable (for now) due to the broad definition of `ArrayLike`
```